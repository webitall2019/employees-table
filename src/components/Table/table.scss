.table{
    margin-bottom: 40px;
    border: 3px solid #000;
    color: #000;
    font-weight: 500;
    &__header{
        .row{
            background-color: rgba(100, 98, 98, 0.466);
        }
    }
    .row{
        display: flex;
        justify-content: space-between;
        .cell{
            width: 25%;
            padding: 5px 10px;
            text-transform: uppercase;
            &:nth-child(3), &:nth-child(4){
                max-width: 10%;
            }
            @media screen and (max-width: 425.1px) {
                padding: 5px;
                &:nth-child(3), &:nth-child(4){
                    max-width: 12%;
                }
            }
        }
        &:nth-child(even){
            background-color: #dfd7d7;
        }
    }
    .cell:not(:first-child){
        border-left: solid;
    }
    @media screen and (max-width: 425.1px) {
        font-size: 11px;
      
    }
}