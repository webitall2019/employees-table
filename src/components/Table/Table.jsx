import { get, map, keys } from "lodash";
import React from "react";
import { useSelector } from "react-redux";
import { useGetData } from "../../hooks/useGetData";

import "./table.scss";

const Table = () => {
  useGetData("./api/employees.json");
  const users = useSelector((state) => state.users);
  const titles = get(users, [0], []);

  const titlesRender = (titles) => {
    return map(keys(titles), (title) => {
      return (
        <div className="cell" key={title}>
          {title}
        </div>
      );
    });
  };
  return (
    <div className="table">
      <div className="table__header">
        <div className="row" data-testid="titleRender">
          {titlesRender(titles)}
        </div>
      </div>
      <div className="table__body">
        {map(users, (el, index) => {
          const genderLetter = get(el.gender, [0], "");

          return (
            <div className="row" key={`${el}_${index}`}>
              <div className="cell">
                <div className="name">{el.name}</div>
              </div>
              <div className="cell">
                <div className="job">{el.jobTitle}</div>
              </div>
              <div className="cell">
                <div className="tenure">{el.tenure}</div>
              </div>
              <div className="cell">
                <div className="gender">{genderLetter}</div>
              </div>
            </div>
          );
        })}
      </div>
    </div>
  );
};
export default Table;
