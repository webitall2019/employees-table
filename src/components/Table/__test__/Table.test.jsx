import React from "react";
import { render, screen } from "@testing-library/react";
import { act } from "react-dom/test-utils";
import "@testing-library/jest-dom";
import Table from "../Table";
import { Provider } from "react-redux";
import store from "../../../redux/store/store";

describe(" <Table/> component", () => {
  it("check redux", async () => {
    render(
      <Provider store={store}>
        <Table />
      </Provider>
    );

    act(async () => {
      await expect(screen.queryByText(/name/i).toBeInTheDocument());
      await expect(screen.queryByText(/jobtitle/i).toBeInTheDocument());
      await expect(screen.queryByText(/tenure/i).toBeInTheDocument());
      await expect(screen.queryByText(/gender/i).toBeInTheDocument());
    });
  });

  it("should match snapshot WHEN rendered", () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Table />
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
