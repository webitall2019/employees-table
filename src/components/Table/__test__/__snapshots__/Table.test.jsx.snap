// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` <Table/> component should match snapshot WHEN rendered 1`] = `
<DocumentFragment>
  <div
    class="table"
  >
    <div
      class="table__header"
    >
      <div
        class="row"
        data-testid="titleRender"
      />
    </div>
    <div
      class="table__body"
    />
  </div>
</DocumentFragment>
`;
